### ISSUE TITLE
[Other]: Why JEI's performance is so much worse?

### ISSUE THREAD

**Issue by TheAirBlow** [mezz/JustEnoughItems#3184]
### Other

I've noticed that since 1.12.2, JEI's performace severely declined. I've even noticed that on newer modpacks the column size was decreased by a lot, and even with x2 less items rendered it is still much worse than 1.12.2.

Switching pages causes the game to freeze for multiple seconds each time, so does search. Opening the inventory causes my FPS to drop from 60 to 15.

I have a 4 core 8 thread CPU, boost up to 3.7 gHZ.
Has a dedicated GPU which I have forced MC to use in Prism Launcher.

I'm using Arch Linux, all packages updated very recently. It is a laptop - set to performance mode, always charging.

RAM allocated: 6144 MB, which always did it for me. The modpack isn't *that* large - it isn't even near 300+.

Extra: I didn't add Optifine, nor Rubidium and it's addons.

**Comment by P3rf3ctXZer0** [comment]
I am here for the same reason glad someone besides myself noticed it too.

**Comment by TheAirBlow** [comment]
Opening a page with a bunch of blocks and items causes me to get the slowest FPS, but it works fine on 1.12.2. Any reason on why so?

**Comment by P3rf3ctXZer0** [comment]
> Opening a page with a bunch of blocks and items causes me to get the slowest FPS, but it works fine on 1.12.2. Any reason on why so?

I am starting to think JEI is a part but not the cause of the lag. I am now able to get it to happen with just E I go from 60 FPS to 20 FPS in inventory and I go from 60 FPS to 11 FPS in refined storage menu. I am launching my own investigation.

**Comment by P3rf3ctXZer0** [comment]
It is not JEI its this for me @TheAirBlow https://github.com/refinedmods/refinedstorage/issues/3496

**Comment by TheAirBlow** [comment]
> It is not JEI its this for me @TheAirBlow [refinedmods/refinedstorage#3496](https://github.com/refinedmods/refinedstorage/issues/3496)

Fortunately I'm not stupid. This happens in any GUI, thus I reported it here.

**Comment by mezz** [comment]
Thanks for the report!
If you restrict the rows/columns of JEI to match the creative menu, it should be just about the same.
I am just rendering items, nothing fancy. Same goes for refined storage, it's just showing a lot of items.

I used to have some rendering optimizations in order to improve the performance, but it would frequently run into mysterious issues with mods or Optifine so I removed them. Unfortunately the only perfectly safe way to render items is to do it in the most "normal" way possible.
